source.php

```diff
@@ -1289,3 +1291,54 @@ function rtsocial_save_meta_box_data( $post_id ) {
 	// Update the meta field in the database.
 	update_post_meta( $post_id, '_rtsocial_visibility', $my_data );
 }
+
+/**
+ * Display number of shares using WordPress HTTP API
+ */
+function rtss_wp_get_shares() {
+	if ( isset( $_GET['security'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$security = sanitize_text_field( wp_unslash( $_GET['security'] ) );
+	} else {
+		$security = '';
+	}
+
+	if ( isset( $_GET['post_id'] ) ) {
+		$post_id = sanitize_text_field( wp_unslash( $_GET['post_id'] ) );
+	} else {
+		$post_id = '';
+	}
+
+	check_ajax_referer( 'rts_media_' . $post_id, 'security' );
+
+	$options      = get_option( 'rtsocial_plugin_options' );
+	$cache_key    = 'rtss_fb' . $post_id;
+	$access_token = $options['fb_access_token'];
+	$count        = get_transient( $cache_key ); // try to get value from WordPress cache.
+
+	if ( ! $access_token ) {
+		$count = false;
+	}
+
+	// if no value in the cache.
+	if ( false === $count || 0 === count ) {
+		$response = wp_remote_get(
+			add_query_arg(
+				array(
+					'id'           => rawurlencode( get_permalink( $post_id ) ),
+					'access_token' => $access_token,
+					'fields'       => 'engagement',
+				),
+				'https://graph.facebook.com/v3.0/'
+			)
+		);
+		$body     = json_decode( $response['body'] );
+		$count    = intval( $body->engagement->share_count );
```

Comment:
@mangesh Please check if we need error handling or pre-check if share_count exists in response.

Reply:
Fixed